src/main/java/com/onlinebookshop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package com.onlinebookshop.service.impl;
+
+import com.onlinebookshop.dto.cartitem.CartItemRequestDto;
+import com.onlinebookshop.dto.cartitem.CartItemResponseDto;
+import com.onlinebookshop.dto.cartitem.CartItemUpdateRequestDto;
+import com.onlinebookshop.dto.shoppingcart.ShoppingCartDto;
+import com.onlinebookshop.exception.EntityNotFoundException;
+import com.onlinebookshop.mapper.CartItemMapper;
+import com.onlinebookshop.mapper.ShoppingCartMapper;
+import com.onlinebookshop.model.CartItem;
+import com.onlinebookshop.model.ShoppingCart;
+import com.onlinebookshop.model.User;
+import com.onlinebookshop.repository.cartitem.CartItemRepository;
+import com.onlinebookshop.repository.shoppingcart.ShoppingCartRepository;
+import com.onlinebookshop.service.BookService;
+import com.onlinebookshop.service.ShoppingCartService;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemMapper cartItemMapper;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final BookService bookService;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public ShoppingCartDto getShoppingCart() {
+        ShoppingCartDto shoppingCartDto = shoppingCartMapper.toDto(getCartOfCurrentUser());
+        shoppingCartDto.setCartItems(getCartOfCurrentUser().getCartItems().stream()
```

Comment:
why are you calling `getCartOfCurrentUser()` 2 times? Considering that you method is not marked Transactional it's 2 queries to the DB

Reply:
not resolved